providers/providers.go

```diff
@@ -103,16 +110,27 @@ func (pm *ProviderManager) getProvSet(k []byte) (*providerSet, error) {
 	}
 
 	if len(pset.providers) > 0 {
-		pm.providers.Add(string(k), pset)
+		pm.cache.Add(string(k), pset)
 	}
 
 	return pset, nil
 }
 
+// Load local providers into cache
```

Comment:
This function doesn't interact with the cache...or am I missing it?

Reply:
yeah, the name is so misleading. There is the func above that uses this one to load the records (pull from datastore) so that the other func can load them into cache. Updated comment in my next commit